R/qenv-get_code.R

```diff
@@ -23,13 +24,19 @@ setGeneric("get_code", function(object, deparse = TRUE) {
 
 #' @rdname get_code
 #' @export
-setMethod("get_code", signature = "qenv", function(object, deparse = TRUE) {
+setMethod("get_code", signature = "qenv", function(object, deparse = TRUE, names = NULL) {
   checkmate::assert_flag(deparse)
-  if (deparse) {
-    format_expression(object@code)
+  code <- if (!is.null(names)) {
+    get_code_dependency(object, names)
   } else {
     object@code
   }
+  if (code[1] == "") code <- code[-1]
```

Comment:
Why is  `code[1]` `""`, actually, when the object is instantiated with `character(0)`?

```
> q <- new_qenv()
> q@code
[1] ""
> q <- new_qenv(new.env(), code = character(0))
> q@code
[1] ""
```

Reply:
no idea : ) it's gonna be a painful edge case, but solvable

Reply:
https://github.com/insightsengineering/teal.code/pull/146/commits/b619c3b9eca10d5fae0c8499cbe635ac44d97c9d

Reply:
fc711381452800920484f2f5e6047c6d79f1a073

```r
> q <- new_qenv()
> q@code
character(0)
> 
> q <- new_qenv(new.env(), code = character(0))
> q@code
character(0)
```